src/transaction/types.ts

```diff
@@ -62,7 +62,7 @@ export type TransactionProviderReact = {
   children: ReactNode; // The child components to be rendered within the provider component.
   contracts: ContractFunctionParameters[]; // An array of contract function parameters provided to the child components.
   onError?: (e: TransactionError) => void; // An optional callback function that handles errors within the provider.
-  onSuccess?: (response: TransactionResponse) => void; // An optional callback function that exposes transaction hash
+  onSuccess?: (response: TransactionReceipt | TransactionReceipt[]) => void; // An optional callback function that exposes transaction hash
```

Comment:
A few questions:
1. why two types? I think we should lean towards `TransactionReceipt[]`.
2. the return value from `onSuccess` should probably be an object. And have `transactionReceipts` as part of it. Just because it will keep the door open for potential future things.

Reply:
i made it so that if there is only one `transactionReceipt` (i.e for smart wallet) we would return just the receipt instead of an array with one receipt inside but agree that your approach would simplify things. updating now!

Reply:
thank you